deformable_manipulation_experiment_params/include/deformable_manipulation_experiment_params/ros_params.hpp

```diff
@@ -82,67 +83,34 @@ namespace smmap
         }
     }
 
+    /**
+     *  Maps the ros param "task_type" into an enum TaskType
+     */
     inline TaskType GetTaskType(ros::NodeHandle& nh)
     {
         const std::string task_type = ROSHelpers::GetParamRequired<std::string>(nh, "task_type", __func__).GetImmutable();
 
-        if (task_type.compare("rope_cylinder_coverage") == 0)
-        {
-            return TaskType::ROPE_CYLINDER_COVERAGE;
-        }
-        else if (task_type.compare("rope_cylinder_coverage_two_grippers") == 0)
-        {
-            return TaskType::ROPE_CYLINDER_COVERAGE_TWO_GRIPPERS;
-        }
-        else if (task_type.compare("cloth_cylinder_coverage") == 0)
-        {
-            return TaskType::CLOTH_CYLINDER_COVERAGE;
-        }
-        else if (task_type.compare("cloth_table_coverage") == 0)
-        {
-            return TaskType::CLOTH_TABLE_COVERAGE;
-        }
-        else if (task_type.compare("cloth_colab_folding") == 0)
-        {
-            return TaskType::CLOTH_COLAB_FOLDING;
-        }
-        else if (task_type.compare("cloth_wafr") == 0)
-        {
-            return TaskType::CLOTH_WAFR;
-        }
-        else if (task_type.compare("cloth_single_pole") == 0)
-        {
-            return TaskType::CLOTH_SINGLE_POLE;
-        }
-        else if (task_type.compare("cloth_wall") == 0)
-        {
-            return TaskType::CLOTH_WALL;
-        }
-        else if (task_type.compare("cloth_double_slit") == 0)
-        {
-            return TaskType::CLOTH_DOUBLE_SLIT;
-        }
-        else if (task_type.compare("rope_maze") == 0)
-        {
-            return TaskType::ROPE_MAZE;
-        }
-        else if (task_type.compare("rope_drag_along_table") == 0)
-        {
-            return TaskType::ROPE_DRAG_ALONG_TABLE;
+        std::unordered_map<std::string, TaskType> task_map{
```

Comment:
May want a newline here for consistency; using brace initialization breaks my own personal logic for when to use newlines, because the braces are not really control flow / code block (kind of they still are) seperators.